internal/errgrouputil/builder.go

```diff
@@ -0,0 +1,54 @@
+// Package errgrouputil contains methods for working with errgroup code.
+package errgrouputil
+
+import (
+	"context"
+	"runtime"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/pomerium/pomerium/pkg/slices"
+)
+
+// BuilderFunc is a function that builds a value of type T
+type BuilderFunc[T any] func(ctx context.Context) (*T, error)
+
+// Build builds a slice of values of type T using the provided builders concurrently
+// and returns the results and any errors.
+func Build[T any](
+	ctx context.Context,
+	builders ...BuilderFunc[T],
```

Comment:
I'm not entirely sure about this API. I'm wondering if there's an alternative where we wouldn't need to allocate this entire slice of functions up front.

Have you considered something more like this?
```go
func Build[I, O any](ctx context.Context, f func(I) (O, error))
```

Reply:
agree its not the most elegant interface. i primarily wanted to concentrate the logic of concurrent building (and the magic formula of SetLimit(runtime.NumCPU()/2+1)) somewhere in one place for the config building speedup in v0.24

we can certainly iterate on a design that looks cleaner, as we still seem to have a few places where we can build heavy configs in parallel.